rust/chains/hyperlane-ethereum/src/mailbox.rs

```diff
@@ -307,111 +301,27 @@ where
-        Ok(count)
-    }
-
-    #[instrument(skip(self))]
-    async fn delivered(&self, id: H256) -> ChainResult<bool> {
-        Ok(self.contract.delivered(id.into()).call().await?)
-    }
-
-    #[instrument(skip(self))]
-    async fn latest_checkpoint(&self, maybe_lag: Option<NonZeroU64>) -> ChainResult<Checkpoint> {
-        let base_call = self.contract.latest_checkpoint();
-        let call_with_lag = match maybe_lag {
-            Some(lag) => {
-                let tip = self
-                    .provider
-                    .get_block_number()
-                    .await
-                    .map_err(ChainCommunicationError::from_other)?
-                    .as_u64();
-                base_call.block(tip.saturating_sub(lag.get()))
-            }
-            None => base_call,
-        };
-        let (root, index) = call_with_lag.call().await?;
-        Ok(Checkpoint {
-            mailbox_address: self.address(),
-            mailbox_domain: self.domain.id(),
-            root: root.into(),
-            index,
-        })
-    }
-
-    #[instrument(skip(self))]
-    #[allow(clippy::needless_range_loop)]
-    async fn tree(&self, lag: Option<NonZeroU64>) -> ChainResult<IncrementalMerkle> {
-        let lag = lag.map(|v| v.get()).unwrap_or(0).into();
-
-        // use consistent block for all storage slot or view calls to prevent
-        // race conditions where tree contents change between calls
-        let fixed_block_number = self
+        let lag = maybe_lag.map(|v| v.get()).unwrap_or(0).into();
+        let fixed_block_number: BlockNumber = self
             .provider
             .get_block_number()
             .await
             .map_err(ChainCommunicationError::from_other)?
             .saturating_sub(lag)
             .into();
 
-        let expected_root = self
-            .contract
-            .root()
-            .block(fixed_block_number)
-            .call()
-            .await?
-            .into();
-
-        // TODO: migrate to single contract view call once mailbox is upgraded
-        // see https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/2250
-        // let branch = self.contract.branch().block(block_number).call().await;
-
-        let mut branch = [H256::zero(); TREE_DEPTH];
-
-        for index in 0..TREE_DEPTH {
-            let slot = U256::from(MERKLE_TREE_CONTRACT_SLOT) + index;
-            let mut location = [0u8; 32];
-            slot.to_big_endian(&mut location);
-
-            branch[index] = self
-                .provider
-                .get_storage_at(
-                    self.contract.address(),
-                    location.into(),
-                    Some(fixed_block_number),
-                )
-                .await
-                .map(Into::into)
-                .map_err(ChainCommunicationError::from_other)?;
-        }
-
-        let count = self
+        let nonce = self
             .contract
-            .count()
+            .nonce()
```

Comment:
fwiw this is off by one from what it used to be

Reply:
hm, I think this could confuse people comparing sealevel and evm metrics, but otherwise shouldn't functionally change anything?

Reply:
what does this mean? i'm missing the off by one bit

Reply:
like mailbox nonce is count - 1

Reply:
let me know if I should subtract one or not

Reply:
resolving this as there's no off-by-one difference